### PR TITLE
Feat/rpc refactor

### PR DESCRIPTION
**Pull request by mod**

### CODE REVIEW

File: protocols/directfund/request.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const DirectFundRequestMethod = "direct_fund"
+
+type DirectFundRequest struct {
+	CounterParty      string                   `msg:"counter_party"`
+	ChallengeDuration uint32                   `msg:"challenge_duration"`
+	Outcome           []shared.SingleAssetExit `msg:"outcome"`
+	AppDefinition     string                   `msg:"app_definition"`
+	AppData           []byte                   `msg:"app_data"`
+	Nonce             uint64                   `msg:"nonce"`
+}
```
Comment (mod) [inline protocols/directfund/request.go]: what is this encoding for?

Comment (calj) [reply protocols/directfund/request.go]: this is to serialize the request using msg pack

File: channel/state/outcome/allocation.go [hunk 2]
```diff
@@ -18,10 +18,10 @@ const (
...
 // Allocation declares an Amount to be paid to a Destination.
 type Allocation struct {
-	Destination    types.Destination // Either an ethereum address or an application-specific identifier
-	Amount         *types.Uint256    // An amount of a particular asset
-	AllocationType AllocationType    // Directs calling code on how to interpret the allocation
-	Metadata       []byte            // Custom metadata (optional field, can be zero bytes). This can be used flexibly by different protocols.
+	Destination    types.Destination `json:"destination"`     // Either an ethereum address or an application-specific identifier
+	Amount         *types.Uint256    `json:"amount"`          // An amount of a particular asset
+	AllocationType AllocationType    `json:"allocation_type"` // Directs calling code on how to interpret the allocation
+	Metadata       []byte            `json:"metadata"`        // Custom metadata (optional field, can be zero bytes). This can be used flexibly by different protocols.
```
Comment (mod) [inline channel/state/outcome/allocation.go:24]: this seems totally irrelevant, we are not passing this information over the network
